src/main/java/com/terraformersmc/terrestria/feature/tree/foliageplacers/PredictiveSpruceFoliagePlacer.java

```diff
@@ -24,15 +23,15 @@
 	// PredictiveSpruceFoliagePlacer.
 	public static final Codec<PredictiveSpruceFoliagePlacer> CODEC = RecordCodecBuilder.create(instance ->
 		fillFoliagePlacerFields(instance).and(
-				UniformIntDistribution.createValidatedCodec(0, 16, 12)
+				IntProvider.createValidatingCodec(0, 16)
```

Comment:
Not sure where this 12 went. Do we know if something is making up for whatever it did?

Reply:
I can't find any alternative vanilla function for 1.17. Should I just port the 1.16 `createValidatedCodec`?

Reply:
So I have searched around the vanilla classes a bit to find out how Mojang handled this; and it seems like they just added the last two arguments together when switching to `IntProvider`:
![image](https://user-images.githubusercontent.com/48808497/148246207-88a8def3-8dc1-48d9-8dd7-1a4088a55945.png)
![image](https://user-images.githubusercontent.com/48808497/148246213-27f8140d-1532-4cd7-a6fe-7f294020c47c.png)

Reply:
Makes sense in a mojang type of non-sensical way